app/app.go

```diff
@@ -934,6 +934,18 @@ func (app *App) InitChainer(ctx sdk.Context, req abci.RequestInitChain) abci.Res
 	app.UpgradeKeeper.SetModuleVersionMap(ctx, app.mm.GetVersionMap())
 	res := app.mm.InitGenesis(ctx, app.appCodec, genesisState)
 
+	// Everything needed for the genesis bridge data should be set during the InitGenesis call,
+	// so we query it after and return it in InitChainResponse.
+	genesisBridgeData, _, err := app.HubGenesisKeeper.PrepareGenesisBridgeData(ctx)
+	if err != nil {
+		panic(fmt.Errorf("failed to prepare genesis bridge data on InitChain: %w", err))
+	}
+	bz, err := tmjson.Marshal(genesisBridgeData)
```

Comment:
why not proto marshaling? it more suitable here

Reply:
we need to pass this proto from rdk to the hub. so we will end up having multiple conversions:
rdk -> proto -> rdk -> hub -> query hub

plus rdk and the hub use different proto generators and different sdk versions, this may lead to additional issues.

i use json because it has less conversions and because it's much more version-agnostic:
rdf -> json -> hub -> query hub

btw app state in InitChain is also passed in json